presto-main/src/main/java/com/facebook/presto/operator/ByteArrayUtils.java

```diff
@@ -0,0 +1,205 @@
+        }
+
+        for (int position = positionsOffset; position < positionsOffset + batchSize; position++) {
+            // Note that positions is logical row number. It may not start from 0 in the values array.
+            unsafe.putLong(target, (long) targetIndex + ARRAY_BYTE_BASE_OFFSET, values[position + offsetBase]);
+            targetIndex += ARRAY_LONG_INDEX_SCALE;
+        }
+        return targetIndex;
+    }
+
+    // Sequential copy
+    public static int writeLongsWithNulls(byte[] target, int targetIndex, long[] values, boolean[] nulls, int positionsOffset, int batchSize, int offsetBase)
+    {
+        if (target.length < targetIndex - ARRAY_BYTE_BASE_OFFSET + batchSize * ARRAY_LONG_INDEX_SCALE || targetIndex < 0) {
+            throw new IndexOutOfBoundsException();
+        }
+
+        for (int i = positionsOffset; i < positionsOffset + batchSize; i++) {
+            // Note that positions is logical row number. It may not start from 0 in the values array.
+            int position = i + offsetBase;
+            unsafe.putLong(target, (long) targetIndex, values[position]);
+            if (!nulls[position]) {
+                targetIndex += ARRAY_LONG_INDEX_SCALE;
+            }
+        }
+        return targetIndex;
+    }
+
+    public static void writeByte(byte[] target, int targetIndex, byte value)
+    {
+        unsafe.putByte(target, (long) targetIndex + ARRAY_BYTE_BASE_OFFSET, value);
+    }
+
+    public static void writeShort(byte[] target, int targetIndex, short value)
+    {
+        unsafe.putShort(target, (long) targetIndex + ARRAY_BYTE_BASE_OFFSET, value);
+    }
+
+    public static void writeInt(byte[] target, int targetIndex, int value)
+    {
+        unsafe.putInt(target, (long) targetIndex + ARRAY_BYTE_BASE_OFFSET, value);
+    }
+
+    public static void writeLong(byte[] target, int targetIndex, long value)
+    {
+        unsafe.putLong(target, (long) targetIndex + ARRAY_BYTE_BASE_OFFSET, value);
+    }
+
+    public static int writeSlice(byte[] target, int targetIndex, Slice slice)
+    {
+        unsafe.copyMemory(slice.getBase(), slice.getAddress(), target, (long) targetIndex + ARRAY_BYTE_BASE_OFFSET, slice.length());
+        return targetIndex + slice.length();
+    }
+
+    public static int fill(byte[] target, int targetIndex, int length, byte value)
+    {
+        unsafe.setMemory(target, targetIndex + ARRAY_BYTE_BASE_OFFSET, length, value);
+        return targetIndex + length;
+    }
+
+    public static int copyBytes(byte[] target, int targetIndex, byte[] source, int sourceIndex, int length)
+    {
+        // The performance of one copy and two copies (one big chunk at  bytes boundary + else) from Slice are about the same.
+        unsafe.copyMemory(source, (long) sourceIndex + ARRAY_BYTE_BASE_OFFSET, target, (long) targetIndex + ARRAY_BYTE_BASE_OFFSET, length);
+        return targetIndex + length;
+    }
+
+    static void writeLengthPrefixedString(SliceOutput output, String value)
```

Comment:
`private static void`

Reply:
Also this appears unused

Reply:
writeLengthPrefixedString() is used by all BlockEncodingBuffers in serializeTo() method. It can be package private but since other methods in ByteArrayUtils are public, I made this public too.